templates/index.html

```diff
@@ -214,7 +230,7 @@ <h1>E-bike routing Metro Manila</h1>
 
         // Directions-from and Directions-to listeners
         document.getElementById('directions-from').addEventListener('click', function () {
-            sourceCoordinates = clickCoordinates    
+            sourceCoordinates = clickCoordinates
             contextMenu.style.display = 'none'
             // Add marker here
```

Comment:
First check if source marker exists, then remove it. After that, add the source marker